cluster-install/install.sh

```diff
@@ -2111,14 +2111,16 @@ step_apply_resources() {
     # Apply CRs (if we have any)
     if [ -f "$crs_file_path" ]; then
         logdebug "apply CRs"
-        if grep -q "AstraConnector" $crs_file_path; then
-            logdebug "delete previous astraconnect if it exists"
-            if ! is_dry_run; then
+        if ! is_dry_run; then
+            if grep -q "AstraConnector" $crs_file_path; then
+                logdebug "delete previous astraconnect if it exists"
                 # Operator doesn't change the astraconnect spec automatically so we need to delete it first (if it exists)
                 kubectl delete -n "$(get_connector_namespace)" deploy/astraconnect &> /dev/null
-                output="$(kubectl apply -f "$crs_file_path")"
```

Comment:
This line is where we apply CRs, and it was being skipped with `COMPONENTS=TRIDENT_ONLY` since it was hidden behind the 
```
if grep -q "AstraConnector" $crs_file_path; then
```
condition, which will always return false with TRIDENT_ONLY.